docs/source/common/optimizers.rst

```diff
@@ -21,46 +21,101 @@ Manual optimization
 For advanced research topics like reinforcement learning, sparse coding, or GAN research, it may be desirable
 to manually manage the optimization process. To do so, do the following:
 
-* Disable automatic optimization in Trainer:  Trainer(automatic_optimization=False)
+* Disable automatic optimization in Trainer:  Trainer(automatic_optimization=False) or override your LightningModule ``automatic_optimization`` property to False
 * Drop or ignore the optimizer_idx argument
-* Use `self.manual_backward(loss)` instead of `loss.backward()` to automatically scale your loss
+* Use `self.manual_backward(loss)` instead of `loss.backward()`.
+
+.. note:: This is only recommended for experts who need ultimate flexibility. Lightning will handle only precision and accelerators logic. The users are left with zero_grad, accumulated_grad_batches, model toggling, etc..
+
+.. note:: Before 1.2, ``optimzer.step`` was calling ``zero_grad`` internally. From 1.2, it is left to the users expertize.
+
+.. note:: To perform ``accumulate_grad_batches`` with one optimizer, you can do as such.
 
 .. code-block:: python
 
-    def training_step(self, batch, batch_idx, optimizer_idx):
+    def training_step(batch, batch_idx, optimizer_idx):
+        opt = self.optimizers()
+
+        def closure()
+            loss = self.compute_loss(batch)
+            self.manual_backward(loss)
+
+        opt.step(closure=closure)
+
+        if batch_idx % 2 == 0:
+            opt.zero_grad()
+
+
+.. note:: ``self.optimizers()`` will return ``LightningOptimizer`` objects. You can access your own optimizer with ``optimizer.optimizer``. However, if you use our own accelerator to perform a step, Lightning won't be able to support accelerators and precision for you.
+
+
+.. code-block:: python
```

Comment:
is this part of the note? the p[probably add move right,,,

Reply:
Done !